src/utils/test/cycles.test.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    assertEquals,
+    assertFalse
+} from "https://deno.land/std@0.160.0/testing/asserts.ts"
+
+import {cycleWith, dependencyOrderOf} from "../cycles.ts"
+
+
+const collatz = (n: number) => n % 2 === 0 ? n/2 : (3 * n + 1)/2
```

Comment:
What is the purpose of this?

Reply:
Purely for testing the algorithmic stuff - I'll see if I can replace/remove/simplify it.